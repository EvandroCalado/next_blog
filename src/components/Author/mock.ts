export default {
  title: 'Evandro Calado',
  resume: 'Desenvolvedor Frontend',
  image: 'https://avatars.githubusercontent.com/u/110628201?v=4',
  alt: 'Evandro Calado',
};
