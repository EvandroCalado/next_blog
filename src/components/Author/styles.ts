import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 3rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border: 3px solid ${theme.colors.secondary};
    border-radius: 50%;
  `}
`;

export const Container = styled.div`
`;

export const Title = styled.h2`
${({ theme }) => css`
font-size: ${theme.font.sizes.xlarge};
`}
`;

export const Paragraph = styled.p`
`;
