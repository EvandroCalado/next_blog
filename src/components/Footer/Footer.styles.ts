import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};
    border-top: 2px solid ${theme.colors.lightGray};
    text-align: center;
  `}
`;
