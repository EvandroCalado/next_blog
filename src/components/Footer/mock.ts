export default {
  footer:
    '<p>Feito por <a href="https://github.com/EvandroCalado" target="_blanck"  >Evandro Calado</a></p>',
};
