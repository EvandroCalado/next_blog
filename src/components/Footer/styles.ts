import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: ${theme.spacings.large} 0;
    border-top: 2px solid ${theme.colors.lightGray};
    margin-top: ${theme.spacings.xxlarge};
  `}
`;

export const Container = styled.div`
  ${({ theme }) => css`
  display: flex;
  align-items: center;
  justify-content: center;
  
  svg {
    width: 5rem;
    margin-left: 1rem;
    color: ${theme.colors.secondary};
  }

  a {
    opacity: 1;
    transition: all 0.3s ease-in-out;
    cursor: pointer;
  }

  a:hover {
    filter: brightness(60%);
  }
  `}
  `;

export const Link = styled.div`
    ${({ theme }) => css`
      span {
        color: ${theme.colors.secondary};
        font-weight: 600;
      }
    `}
  `;
