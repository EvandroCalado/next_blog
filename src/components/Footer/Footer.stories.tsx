import Footer, { FooterProps } from './Footer';
import { Meta, StoryFn } from '@storybook/react';

import mock from './mock';

export default {
  title: 'Footer',
  component: Footer,
  args: mock,
} as Meta;

export const Template: StoryFn<FooterProps> = (args) => (
  <div>
    <Footer {...args} />
  </div>
);
