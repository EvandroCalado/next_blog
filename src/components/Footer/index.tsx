import * as Styled from './styles';

export type FooterProps = {
  footer: string;
};

const Footer = ({ footer }: FooterProps) => {
  return (
    <Styled.Wrapper
      dangerouslySetInnerHTML={{ __html: footer }}
      data-testid="footer-wrapper"
    />
  );
};

export default Footer;
