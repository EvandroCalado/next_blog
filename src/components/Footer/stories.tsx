import Footer, { FooterProps } from '.';
import { Meta, StoryFn } from '@storybook/react';

import mock from './mock';

export default {
  title: 'Footer',
  component: Footer,
  args: mock,
} as Meta;

export const Template: StoryFn<FooterProps> = (args) => (
  <div>
    <Footer {...args} />
  </div>
);
