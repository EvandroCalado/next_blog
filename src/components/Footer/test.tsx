import Footer from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

import mock from './mock';

describe('<Footer />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Footer {...mock} />);
    expect(screen.getByRole('link')).toBeInTheDocument();
    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      'https://github.com/EvandroCalado',
    );
    expect(container).toMatchSnapshot();
  });
});
