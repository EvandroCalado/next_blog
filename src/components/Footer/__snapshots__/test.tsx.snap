// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 1`] = `
.c0 {
  padding: 3.2rem;
  margin-top: 3.2rem;
  border-top: 2px solid #dddddd;
  text-align: center;
}

<div>
  <div
    class="c0"
  >
    <p>
      Feito por 
      <a
        href="https://github.com/EvandroCalado"
        target="_blanck"
      >
        Evandro Calado
      </a>
    </p>
  </div>
</div>
`;
